assembl/graphql/schema.py

```diff
@@ -1021,29 +1135,46 @@ def mutate(root, args, context, info):
         with cls.default_db.no_autoflush:
             subject = args.get('subject')
             body = args.get('body')
-            if subject is None:
-                # TODO for real post, need the same logic than in views/api/post.py
-                subject_entries = [
-                    {'value': u'Proposition', u'locale_code': u'und'}
-                ]
-            else:
-                subject_entries = [
-                    {'value': subject, u'locale_code': u'und'}
-                ]
+            if subject is None: # We apply the same logic than in views/api/post.py::create_post
+                if cls == models.AssemblPost:
+                    if in_reply_to_post:
+                        subject = (
+                            in_reply_to_post.get_title().first_original().value or ''
+                            if in_reply_to_post.get_title() else '')
+                    elif in_reply_to_idea:
+                        subject = (in_reply_to_idea.short_title
+                                   if in_reply_to_idea.short_title else '')
+                    else:
+                        subject = discussion.topic if discussion.topic else ''
+                    if subject is not None and len(subject):
+                        subject = u'Re: ' + restrip_pat.sub('', subject).strip()
+                else:
+                    subject = u'Proposition'
+            subject_entries = [
+                {'value': subject, u'locale_code': u'und'}
+            ]
 
             body_entries = [
                 {'value': body, u'locale_code': u'und'}
             ]
 
             subject_langstring = langstring_from_input_entries(subject_entries)
             body_langstring = langstring_from_input_entries(body_entries)
-            discussion = models.Discussion.get(discussion_id)
-            new_post = cls(
-                discussion=discussion,
-                subject=subject_langstring,
-                body=body_langstring,
-                creator_id=user_id
-            )
+            if in_reply_to_post: # Is there an unified way to do this?
```

Comment:
Can't you just always use parent=in_reply_to_post here? If in_reply_to_post is None, it should keep the parent field to null.